CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 # Defines BitShares library target.
-project( BitShares )
-cmake_minimum_required( VERSION 2.8.12 )
+cmake_minimum_required( VERSION 3.2 FATAL_ERROR )
```

Comment:
Any specific reason for this? We usually avoid updating minimum requirements.

Reply:
Yes, the same as in https://github.com/bitshares/bitshares-fc/pull/121

CMake that supports Visual Studio 2015 Generator.

Reply:
@abitmore @jmjatlanta what's the minimum cmake version we should support in your opinion / to your knowledge of windows/apple/linux build environments?

Reply:
Another update,

The **first CMake** that is usable with the most up to date version of OpenSSL ( 1.0.2r ) is **3.1.2**.

Any version older than that will fail, with problems parsing version information from OpenSSL headers.

I have reworked the `FindPython` and `FindBoost` routines to be more future-proof.

Updates are coming soon.

Reply:
> The first CMake that is usable with the most up to date version of OpenSSL ( 1.0.2r ) is 3.1.2.

That's helpful, thanks!